include/PathUtilities.h

```diff
@@ -0,0 +1,289 @@
+    Boole MEZZ_LIB IsPathAbsolute_Windows(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative.
+    /// @warning This method is imperfect and doesn't check for a faulty path, just that it's not absolute.
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @details This will check if the path is absolute on either Windows or Posix platforms.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is relative, false otherwise.
+    Boole MEZZ_LIB IsPathRelative(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative on the host.
+    /// @warning This method is imperfect and doesn't check for a faulty path, just that it's not absolute.
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @details This will call the version of this method that matches the compiled platform.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered relative on the host platform, false otherwise.
+    Boole MEZZ_LIB IsPathRelative_Host(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative on Posix.
+    /// @warning This method is imperfect and doesn't check for some faults that can occur in a path, but does
+    /// take the extra time to verify the absence of Windows style directory separators (\\).
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered relative on a Posix platform, false otherwise.
+    Boole MEZZ_LIB IsPathRelative_Posix(const StringView ToCheck);
+    /// @brief Gets whether or not a path is relative on Windows.
+    /// @warning This method is imperfect and doesn't check for some faults that can occur in a path, nor does
+    /// it check for proper Windows style directory separators as most uses of filesystem paths in the Windows
+    /// API can tolerate Windows or Posix syle.
+    /// @note A path is relative if it requires using the current working directory to define the location of
+    /// a resource.
+    /// @param ToCheck The path to check.
+    /// @return Returns true if the specified path is considered relative on a Windows platform, false otherwise.
+    Boole MEZZ_LIB IsPathRelative_Windows(const StringView ToCheck);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Path Checks
+
+    /// @brief Gets how many directories deep a path is.
+    /// @param ToCheck The path to get the depth of.
+    /// @param ExitIfNegative If true, the function to return immediately if the depth count becomes negative.
```

Comment:
This is a really simple feature for security, Exiting if Negative, is this new? I don't recall seeing it other places.

Reply:
I don't know.  It's simply something I thought of as a performance optimization in the event I didn't want to above the current directory...which seems like a common use case.  

I'm apprehensive about defaulting the bool, however sane it may seem because making it true ensures you'll only ever get -1 should it go above the current directory, which isn't accurate in the absolute sense, whereas in all other cases you will get a correct answer for the full path. I think it's important that the dev is entirely aware of the choice they are making for correct usage.